overlay/mkcrate.nix

```diff
@@ -167,14 +167,20 @@ let
         echo source = \"registry+''${registry}\" >> Cargo.lock
       fi
       mv Cargo.toml Cargo.original.toml
+      # Remarshal version cannot handle escaped double quotes in table key paths
+      while grep '\[[^"]*"[^\\"]*\\"[^\\"]*\\"[^"]*[^]]*\]' Cargo.original.toml; do
+        sed -i -r 's/\[([^"]*)"([^\\"]*)\\"([^\\"]*)\\"([^"]*)"([^]]*)\]/[\1"\2'"'"'\3'"'"'\4"\5]/g' Cargo.original.toml
+      done;
```

Comment:
The PR looks great overall, but it bothers me that I can't look at this bit of code and immediately reason about what it does 🤔 I feel like I'm being nit-picky though and not sure what we could do about it anyway. Any thoughts on possible ways to make it more legible?

Reply:
Guess I'll draft a comment and explain the hunk at the same time

```shell
# Remarshal was choking on table names of the form:
# [key."cfg(foo = \"a\", bar = \"b\"))".path]
# The regex to find or deconstruct these strings must find, in order, these components:
# open bracket, open quote, open escaped quote, and their closing pairs.
# Because each quoted path can contain multiple quote escape pairs, a loop is employed
# to match the first quote escape, which the sed will replace with a single quote equivalent
# until all escaped quote pairs are replaced.  The grep regex is identical to the sed regex
# but does not destructure the match into groups for restructuring in the replacement.
```

It probably can't be more legible.  This is a problem of writing a regex that has to match a lot of special regex characters ('[' and '\' and ']'.  The only nice thing I could do was consume each inverse up to the match so there's no back reference etc.  Up to writing a parser, I think this about what we can expect.